src/librbd/api/Mirror.cc

```diff
@@ -1097,6 +1098,33 @@ int Mirror<I>::mode_set(librados::IoCtx& io_ctx,
   return 0;
 }
 
+template <typename I>
+int Mirror<I>::uuid_get(librados::IoCtx& io_ctx, std::string* mirror_uuid) {
+  CephContext *cct = reinterpret_cast<CephContext *>(io_ctx.cct());
+  ldout(cct, 20) << dendl;
+
+  C_SaferCond ctx;
+  uuid_get(io_ctx, mirror_uuid, &ctx);
+  int r = ctx.wait();
+  if (r < 0 && r != -ENOENT) {
```

Comment:
nit: shouldn't we clear `mirror_uuid` when we got `-ENOENT` and return 0 insted? Just thinking about the case, when a caller supplies a non-empty `mirror_uuid`, internaly `uuid_get` gets `-ENOENT` but returns 0 to the caller, so it is valid her to assume that `mirror_uuid` is overwritten with the valid content.

Reply:
I broke this during an internal refactor to add the async version. It used to just avoid printing the error message on `-ENOENT` but still return the error code. Fixed.